gobblin-compaction/src/main/java/gobblin/compaction/source/CompactionSource.java

```diff
@@ -84,6 +91,135 @@
     return workUnits;
   }
 
+  @Override
+  public WorkUnitStream getWorkunitStream(SourceState state) {
+    service = Executors.newSingleThreadExecutor();
+
+    try {
+      fs = getSourceFileSystem(state);
+      suite = CompactionSuiteUtils.getCompactionSuiteFactory(state).createSuite(state);
+
+      initJobDir(state);
+      copyJarDependencies(state);
+      List<CompactionVerifier> verifiers = suite.getDatasetsFinderVerifiers();
+      DatasetsFinder finder = DatasetUtils.instantiateDatasetFinder(state.getProperties(),
+              getSourceFileSystem(state),
+              DefaultFileSystemGlobFinder.class.getName());
+
+      List<Dataset> datasets = finder.findDatasets();
+      CompactionDatasetIterator iterator = new CompactionDatasetIterator(datasets, verifiers);
+      service.submit(iterator.getDatasetProcessor());
```

Comment:
You probably want multiple data processors right? Note that will also require to transform the `List` into something like a `Spliterator` (although that is only available on java 8)